model/task_history.go

```diff
@@ -166,3 +176,47 @@ func GetOldestMainlineTask(ctx context.Context, opts FindTaskHistoryOptions) (*t
 	}
 	return oldestTask, nil
 }
+
+// getNewerActiveMainlineTask returns the next newer active mainline task, i.e. a more
+// recent activated task than the current task.
+func getNewerActiveMainlineTask(ctx context.Context, t task.Task) (*task.Task, error) {
```

Comment:
If we are explicitly looking for mainline tasks here, we should filter on tasks that have the `RepotrackerVersionRequester` requester since the revision order number for a task means different things depending on the requester.

Reply:
I was a little concerned that it searches in `SystemVersionRequesterTypes` rather than explicitly for the mainline requester, but looking at the code it seems like that won't cause any issues w.r.t revision order numbers.